### PR TITLE
Scenes: Row controls

### LINKED ISSUE #83606: Rows: Enable editing row options
Migrate row controls to scenes.

### PR DESCRIPTION
**Pull request by mdvictor**
Migrate row controls to scenes.

Depends on https://github.com/grafana/scenes/pull/624, https://github.com/grafana/scenes/pull/626, https://github.com/grafana/scenes/pull/625

Fixes https://github.com/grafana/grafana/issues/83606

### CODE REVIEW

File: public/app/features/dashboard-scene/scene/RowRepeaterBehavior.ts [hunk 1]
```diff
@@ -180,8 +180,12 @@ function updateLayout(layout: SceneGridLayout, rows: SceneGridRow[], maxYOfRows:
     const diff = maxYOfRows - firstChildAfterY;
 
     for (const child of childrenAfter) {
-      if (child.state.y! < maxYOfRows) {
```
Comment (mdvictor) [inline public/app/features/dashboard-scene/scene/RowRepeaterBehavior.ts:183]: Not sure if I'm missing something but I think we don't need this if here. There are cases where the child state y is larger than the max y of repeated rows and we do want to offset those as well.

File: public/app/features/dashboard-scene/scene/row-actions/RowActions.tsx [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+    if (!(this.parent instanceof SceneGridRow)) {
+      throw new Error('RowActions must have a SceneGridRow parent');
+    }
+
+    return this.parent;
+  }
+
+  public getDashboard(): DashboardScene {
+    return this._dashboard;
+  }
```
Comment (dprokop) [inline public/app/features/dashboard-scene/scene/row-actions/RowActions.tsx:74]: Don't think we need the private getter `_dashboard` while we have a public `getDashboard` method.

File: public/app/features/dashboard-scene/scene/DashboardScene.tsx [hunk 3]
```diff
@@ -407,6 +407,25 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
     });
   }
 
+  public removeRow(row: SceneGridRow, removePanels: boolean) {
```
Comment (dprokop) [inline public/app/features/dashboard-scene/scene/DashboardScene.tsx:410]: please add a test case for this method :)

File: public/app/features/dashboard-scene/panel-edit/PanelOptions.tsx [hunk 4]
```diff
@@ -15,12 +15,15 @@ interface Props {
...
+  const { panel, repeat } = vizManager.useState();
   const { data } = sceneGraph.getData(panel).useState();
   const { options, fieldConfig } = panel.useState();
 
   // eslint-disable-next-line react-hooks/exhaustive-deps
-  const panelFrameOptions = useMemo(() => getPanelFrameCategory2(vizManager), [vizManager, panel]);
+  const panelFrameOptions = useMemo(
+    () => getPanelFrameCategory2(vizManager, panel, repeat),
+    [vizManager, panel, repeat]
+  );
```
Comment (dprokop) [inline public/app/features/dashboard-scene/panel-edit/PanelOptions.tsx:26]: Why do we need this change?

**Review by dprokop — COMMENTED** [review]
@mdvictor i did a quick review and tets of this draft and it looks good already. Let me know when it's ready for a final review.

File: public/app/features/dashboard-scene/panel-edit/PanelOptions.tsx (hunk 4 above)
Comment (mdvictor) [reply public/app/features/dashboard-scene/panel-edit/PanelOptions.tsx:26]: This is one spot I'm having doubts myself, so if there is a better way to do it I'm open to change it. 

We need these state values to send them later on to `RepeatRowSelect2` and with this refactor we make `RepeatRowSelect2` reusable for row controls as well. Before this, `RepeatRowSelect2` took the `VizPanelManager` and subscribed to state inside, but this cannot be used for row controls in the dashboard scene cuz there is no `VizPanelManager` involved.

**Comment by mdvictor** [comment]
Thank you @dprokop, I'm writing tests as we speak and it should be good for review after that.

File: public/app/features/dashboard-scene/scene/row-actions/RowActions.tsx [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+import { RowOptionsButton } from './RowOptionsButton';
+
+export interface RowActionsState extends SceneObjectState {}
+
+export class RowActions extends SceneObjectBase<RowActionsState> {
+  public constructor(state: RowActionsState) {
+    super({
+      ...state,
+    });
+  }
```
[the remark this answers is not in the document]
Comment (mdvictor) [reply public/app/features/dashboard-scene/scene/row-actions/RowActions.tsx]: Good catch, thanks! 💯

**Review by dprokop — APPROVED** [review]

**Pull request merged**